src/engine/source/builder/builders/windowsHelper.cpp

```diff
@@ -0,0 +1,224 @@
+
+/**
+ * @brief Parse the list of SID, the list must be in the format:
+ *
+ * '%{sid1} %{sid2} %{sid3} ... ' // TODO: Check the format
+ *
+ * Ths function will return a vector with the sids in the same order as the input
+ * or return an empty vector if the input is not valid
+ * @param listSrt String with the list of sids
+ * @return std::vector<std::string>
+ */
+std::vector<std::string> parserListSID(const std::string& listSrt)
+{
+
+    std::vector<std::string> result = base::utils::string::split(listSrt, ' '); // TODO Check format
+
+    for (auto& sid : result)
+    {
+        if (sid.size() > 3)
+        {
+            // Remove header '%{' and tail '}'
+            sid = sid.substr(2, sid.size() - 3);
+        }
+    }
+
+    return result;
+}
+
+} // namespace
+
+namespace builder::internals::builders
+{
+HelperBuilder getWindowsSidListDescHelperBuilder(std::shared_ptr<kvdbManager::IKVDBManager> kvdb,
+                                                 const std::string& kvdbScopeName,
+                                                 std::shared_ptr<schemf::ISchema> schema)
+{
+    // TODO: add shared_ptr validation when tests are updated
+
+    return [=](const std::string& targetField,
+               const std::string& rawName,
+               const std::vector<std::string>& rawParameters,
+               std::shared_ptr<defs::IDefinitions> definitions) -> base::Expression
+    {
+        // Process parameters
+        auto parameters = processParameters(rawName, rawParameters, definitions);
+
+        // Format name
+        const auto name = formatHelperName(targetField, rawName, parameters);
+
+        // Check parameters
+        checkParametersSize(rawName, parameters, 2);
+        checkParameterType(rawName, parameters[0], Parameter::Type::VALUE);
+        checkParameterType(rawName, parameters[1], Parameter::Type::REFERENCE);
+
+        auto kvdbName = parameters[0].m_value;
+        auto sidListRef = parameters[1].m_value;
+
+        if (schema->hasField(sidListRef) && schema->getType(sidListRef) != json::Json::Type::String)
+        {
+            throw std::runtime_error(fmt::format("The reference '{}' is not an string.", sidListRef));
+        }
+        if (schema->hasField(targetField) && schema->getType(targetField) != json::Json::Type::Array)
+        {
+            throw std::runtime_error(fmt::format("The target field '{}' is not an array.", targetField));
+        }
+
+        // Get the kvdb handler
+        auto kbdbRes = kvdb->getKVDBHandler(kvdbName, kvdbScopeName);
+        if (base::isError(kbdbRes))
+        {
+            throw std::runtime_error(
+                fmt::format("Error getting the kvdb handler: {}", base::getError(kbdbRes).message));
+        }
+
+        auto kvdbHandler = base::getResponse<std::shared_ptr<kvdbManager::IKVDBHandler>>(kbdbRes);
+
+        // Get the lists
+        // Account SID Description
+        std::map<std::string, std::string> asdMap;
```

Comment:
improve-desing: Both adsMap and dssMap have the same logic, could repeated code be eliminated?

Reply:
Yes!, done